src/Microsoft.Framework.DependencyInjection/ServiceLookup/ServiceTable.cs

```diff
@@ -47,6 +59,19 @@ public ServiceTable(IEnumerable<ServiceDescriptor> descriptors)
                 }
                 else
                 {
+                    Debug.Assert(descriptor.ImplementationType != null);
+                    var implementationTypeInfo = descriptor.ImplementationType.GetTypeInfo();
```

Comment:
Which test was failing? Not sure I understand the scenario... though, the code in the `else` I think is fine...